base/linalg/tridiag.jl

```diff
@@ -39,7 +48,15 @@ end
 convert{T}(::Type{Matrix}, M::SymTridiagonal{T}) = convert(Matrix{T}, M)
 
 size(A::SymTridiagonal) = (length(A.dv), length(A.dv))
-size(A::SymTridiagonal, d::Integer) = d<1 ? throw(ArgumentError("dimension must be ≥ 1, got $d")) : (d<=2 ? length(A.dv) : 1)
+function size(A::SymTridiagonal, d::Integer)
+    if d < 1
```

Comment:
The ternary means we won't be able to see if the condition `d > 2` is checked/thrown correctly. Although it seems obvious I think it's worth making sure that we are testing every branch. There have been bugs in equivalently "obvious" branches that weren't tested.